nixos/modules/system/boot/loader/limine/limine.nix

```diff
@@ -0,0 +1,155 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib;
+let
+  cfg = config.boot.loader.limine;
+  efi = config.boot.loader.efi;
+  limineInstallConfig = pkgs.writeText "limine-install.json" (
+    builtins.toJSON {
+      nixPath = config.nix.package;
+      efiBootMgrPath = pkgs.efibootmgr;
+      liminePath = cfg.package;
+      efiMountPoint = efi.efiSysMountPoint;
+      fileSystems = config.fileSystems;
+      luksDevices = config.boot.initrd.luks.devices;
+      canTouchEfiVariables = efi.canTouchEfiVariables;
+      maxGenerations = if cfg.maxGenerations == null then 0 else cfg.maxGenerations;
+      hostArchitecture = pkgs.stdenv.hostPlatform.parsed.cpu;
+      timeout = if config.boot.loader.timeout != null then config.boot.loader.timeout else 10;
+      enableEditor = cfg.enableEditor;
+      extraEntries = cfg.extraEntries;
+      additionalFiles = cfg.additionalFiles;
+      useStorePaths = cfg.useStorePaths;
+      validateChecksums = cfg.validateChecksums;
+      panicOnChecksumMismatch = cfg.panicOnChecksumMismatch;
+    }
+  );
+in
+{
+  options.boot.loader.limine = {
+    enable = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Whether to enable the Limine bootloader.
+      '';
+    };
+
+    enableEditor = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to allow editing the boot entries before booting them.
+        It is recommended to set this to false, as it allows gaining root
+        access by passing `init=/bin/sh` as a kernel parameter.
+      '';
+    };
+
+    maxGenerations = mkOption {
+      default = 10;
```

Comment:
IIRC systemd-boot (and probably grub too) does not limit this by default (I suggest verifying that, but I have 200+ entries in mine). Boot partition disk space is probably more than enough for potentially thousands of entries, this is simply just a plain text config file. 10 is nowhere near enough for a default value here.

Reply:
The config file with a bunch of entries won't take up much space but you also have to factor in the kernels and modules that need to be on the ESP. I believe that my original script copied everything without any form of deduplication so it adds up really quick.

Reply:
Ah, that makes sense. We should check how this is handled in other bootloader modules then @programmerlexi

Reply:
A complicating factor is that systemd-boot and grub support ext4, meaning they can mount the Nix store partition on most systems and avoid copying kernels and initrds. Limine by-design only supports FAT and ISO9660, meaning a dedicated boot partition is required even with BIOS boot (unless you somehow managed to put your nix store on a FAT32 partition).